core/go/internal/keymanager/manager.go

```diff
@@ -242,3 +244,34 @@ func (km *keyManager) ReverseKeyLookup(ctx context.Context, dbTX *gorm.DB, algor
 	km.verifierReverseCache.Set(vKey, mapping)
 	return mapping, nil
 }
+
+func (km *keyManager) QueryKeys(ctx context.Context, dbTX *gorm.DB, jq *query.QueryJSON) (keyList []*pldapi.KeyQueryEntry, err error) {
+
+	q := filters.BuildGORM(ctx, jq,
+		dbTX.WithContext(ctx).
+			Table("key_paths"), KeyEntryFilters)
+
+	q.Select(`key_verifiers.identifier IS NOT NULL AS "is_key",` +
+		`k.p IS NOT NULL AS "has_children",` +
+		`key_paths.parent AS "parent",` +
+		`key_paths.index AS "index",` +
+		`key_paths.path AS "path",` +
+		`key_verifiers.type AS "type",` +
+		`key_verifiers.algorithm AS "algorithm",` +
+		`key_verifiers.verifier AS "verifier",` +
+		`key_mappings.wallet AS "wallet",` +
+		`key_mappings.key_handle AS "key_handle"`,
+	)
+
+	q.Joins("LEFT OUTER JOIN key_verifiers ON key_paths.path = key_verifiers.identifier")
+	q.Joins("LEFT OUTER JOIN key_mappings ON key_paths.path = key_mappings.identifier")
+	q.Joins(`LEFT OUTER JOIN (SELECT parent AS "p" from key_paths AS p) AS k ON key_paths.path = k.p`)
```

Comment:
Just a question on how this behaves when there are lots of children

Reply:
Excellent point, we need to introduce distinct to address that scenario. I'll update the code.